src/Yoti.Auth/SignedTimestamp.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+
+namespace Yoti.Auth
+{
+    /// <summary>
+    /// SignedTimestamp is a timestamp associated with a message that has a
+    /// cryptographic signature proving that it was issued by the correct authority
+    /// </summary>
+    public class SignedTimestamp
+    {
+        private readonly Int32 _version;
+        private readonly DateTime _timestamp;
+        private readonly byte[] _messageDigest;
+        private readonly byte[] _chainDigest;
+        private readonly byte[] _chainDigestSkip1;
+        private readonly byte[] _chainDigestSkip2;
+
+        public SignedTimestamp(CompubapiV1.SignedTimestamp protobufSignedTimestamp)
+        {
+            _timestamp = ConvertMicroSecondsSinceEpochToDateTime(protobufSignedTimestamp);
+            _version = protobufSignedTimestamp.Version;
+            _messageDigest = protobufSignedTimestamp.MessageDigest.ToByteArray();
+            _chainDigest = protobufSignedTimestamp.ChainDigest.ToByteArray();
+            _chainDigestSkip1 = protobufSignedTimestamp.ChainDigestSkip1.ToByteArray();
+            _chainDigestSkip2 = protobufSignedTimestamp.ChainDigestSkip2.ToByteArray();
+        }
+
+        private static DateTime ConvertMicroSecondsSinceEpochToDateTime(CompubapiV1.SignedTimestamp protobufSignedTimestamp)
+        {
+            DateTime epoch = new DateTime(1970, 1, 1);
+            long timeInMicroSecondsSinceEpoch = (long)protobufSignedTimestamp.Timestamp;
+            long timeInTicksSinceEpoch = timeInMicroSecondsSinceEpoch * 10;
+            return epoch.Add(new TimeSpan(timeInTicksSinceEpoch));
```

Comment:
Can't you just return a ` new DateTime(timeInMicroSecondsSinceEpoch * 10)` ?

Reply:
As discussed, need to keep it this way to get the correct date, and be clear of what is going on, even if it's a little cumbersome.